tracer/src/Datadog.Trace/Debugger/Expressions/ProbeProcessor.cs

```diff
@@ -122,16 +121,35 @@ public IDebuggerSnapshotCreator CreateSnapshotCreator()
         private void SetExpressions(ProbeDefinition probe)
         {
             // ReSharper disable once PossibleInvalidOperationException
-            _templates = (probe as LogProbe)?.Segments?.Where(seg => seg != null).Select(seg => ToDebuggerExpression(seg).Value).ToArray();
-            _condition = ToDebuggerExpression((probe as LogProbe)?.When);
-            _metric = ToDebuggerExpression((probe as MetricProbe)?.Value);
-            _spanDecorations = (probe as SpanDecorationProbe)?.Decorations?.Where(dec => dec != null).Select(dec =>
+            switch (probe)
             {
-                return new KeyValuePair<DebuggerExpression?, KeyValuePair<string, DebuggerExpression[]>[]>(
-                    ToDebuggerExpression(dec.When),
-                    dec.Tags.Select(tag => new KeyValuePair<string, DebuggerExpression[]>(
-                                        tag.Name, tag.Value.Segments?.Where(seg => seg != null).Select(seg => ToDebuggerExpression(seg).Value).ToArray())).ToArray());
-            }).ToArray();
+                case LogProbe logProbe:
```

Comment:
Since model properties are nullable now, you might want to check if `ToDebuggExpression` returns null (or check for null before call it)

Reply:
I wish we could add nullable enable to all of our models.
I believe it can wait till we decide to do so?